classes/submissionFile/Repository.php

```diff
@@ -843,4 +859,262 @@ protected function getSubmissionFileLogData(SubmissionFile $submissionFile): arr
             'username' => $user?->getUsername(),
         ];
     }
+
+    /**
+     * Add jatsContent for Submission files that correspont to 
+     */
+    public function getJatsFileProps(SubmissionFile $submissionFile, array $genres): array
+    {
+        $fileProps = $this->getSchemaMap()->summarize($submissionFile, $genres);
+        $jatsFileName = Config::getVar('files', 'files_dir') . '/' . $fileProps['path'] .'';
+        $xmlString = file_get_contents($jatsFileName);
+
+        if ($xmlString) {
+            $fileProps['jatsContent'] = $xmlString;
+        }
+
+        return $fileProps;
+    }
+
+    /**
+     * Returns the SubmissionFile, if any, that corresponds to the JATS contents of the given submission/publication
+     */
+    public function getJatsFiles(int $submissionId, int $publicationId): ?SubmissionFile
+    {
+        $this->validateSubmissionAndPublication($submissionId, $publicationId);
+
+        $submissionFile = Repo::submissionFile()
+            ->getCollector()
+            ->filterBySubmissionIds([$this->submission->getId()])
```

Comment:
The `$publicationId` filter implicitly limits by submission ID as well. I'd suggest modifying this function to make `$publicationId` the first parameter, and making the `$submissionId` optional; when it's specified, then this filter is applied as well. That way it can be used optionally as an extra guardrail (and follows a familiar pattern we use elsewhere). And we can do away with extra functions like `validateSubmissionAndPublication`, which can quickly collect and be used inconsistently.

Reply:
I removed the `validateSubmissionAndPublication` function.